crates/driver/src/tests/cases/solver_balance.rs

```diff
@@ -0,0 +1,245 @@
+            })
+        })
+        .collect_vec();
+
+    // Set up the solver.
+    let solver = setup::solver::setup(setup::solver::Config {
+        name: SOLVER_NAME.to_owned(),
+        absolute_slippage: "0".to_owned(),
+        relative_slippage: "0.0".to_owned(),
+        address: hex_address(solver_address),
+        private_key: format!("0x{}", solver_secret_key.display_secret()),
+        solve: vec![setup::solver::Solve {
+            req: json!({
+                "id": "1",
+                "tokens": {
+                    hex_address(sell_token): {
+                        "decimals": null,
+                        "symbol": null,
+                        "referencePrice": "1",
+                        "availableBalance": "0",
+                        "trusted": false,
+                    },
+                    hex_address(buy_token): {
+                        "decimals": null,
+                        "symbol": null,
+                        "referencePrice": "2",
+                        "availableBalance": "0",
+                        "trusted": false,
+                    }
+                },
+                "orders": [
+                    {
+                        "uid": boundary.uid(),
+                        "sellToken": hex_address(sell_token),
+                        "buyToken": hex_address(buy_token),
+                        "sellAmount": sell_amount.to_string(),
+                        "buyAmount": buy_amount.to_string(),
+                        "feeAmount": "0",
+                        "kind": "sell",
+                        "partiallyFillable": false,
+                        "class": "market",
+                        "reward": 0.1,
+                    }
+                ],
+                "liquidity": [],
+                "effectiveGasPrice": "247548525",
+                "deadline": deadline - auction::Deadline::time_buffer(),
+            }),
+            res: json!({
+                "prices": {
+                    hex_address(sell_token): buy_amount.to_string(),
+                    hex_address(buy_token): sell_amount.to_string(),
+                },
+                "trades": [
+                    {
+                        "kind": "fulfillment",
+                        "order": boundary.uid(),
+                        "executedAmount": sell_amount.to_string(),
+                    }
+                ],
+                "interactions": interactions
+            }),
+        }],
+    })
+    .await;
+
+    // Set up the driver.
+    let client = setup::driver::setup(setup::driver::Config {
+        now,
+        file: setup::driver::ConfigFile::Create {
+            solvers: vec![solver],
+            contracts: infra::config::file::ContractsConfig {
+                gp_v2_settlement: Some(settlement.address()),
+                weth: Some(weth.address()),
+            },
+        },
+        geth: &geth,
+    })
+    .await;
+
+    // Transfer out all of the solver's funds.
```

Comment:
Would it be easily doable to verify that we require `current_balance >= settlement_gas * max_gas_price_after_n_blocks`?

Reply:
Nice! Added this (at least, my interpretation of the suggestion) and caught a small issue - we weren't considering the `max_priority_fee_per_gas` in the computation.

Reply:
Yes, basically verify the check you've been working on that the solver has to have enough eth to submit the solution even if the gas price increases by `12.5%` for 10 blocks consecutively.

Reply:
So I changed the E2E test to leave a reasonable amount of Eth (more than enough for a `settle` transaction) but not enough if a gas spike were to happen, and asserted that the balance check works there.

Actually filling blocks to 100% gas usage for 10 consecutive blocks is pretty tricky though...